payroll/models/hr_payslip.py

```diff
@@ -683,61 +688,69 @@ def _sum_salary_rule_category(self, localdict, category, amount):
         return localdict
 
     def _get_employee_contracts(self):
-        return self.env["hr.contract"].browse(
-            self.employee_id._get_contracts(
-                date_from=self.date_from, date_to=self.date_to
-            ).ids
-        )
+        contracts = self.env["hr.contract"]
+        for payslip in self:
+            contracts |= payslip.employee_id._get_contracts(
```

Comment:
Why is the |= needed here?

Reply:
because self is a recordset which may include more than one payslip. When the method returns it will return all the contracts from all the payslips in the recordset, minus duplicates.